daft/runners/ray_tracing.py

```diff
@@ -0,0 +1,666 @@
+"""This module contains utilities and wrappers that instrument tracing over our RayRunner's task scheduling + execution
+
+These utilities are meant to provide light wrappers on top of Ray functionality (e.g. remote functions, actors, ray.get/ray.wait)
+which allow us to intercept these calls and perform the necessary actions for tracing the interaction between Daft and Ray.
+"""
+
+from __future__ import annotations
+
+import contextlib
+import dataclasses
+import json
+import logging
+import pathlib
+import time
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Iterator, TextIO
+
+try:
+    import ray
+except ImportError:
+    raise
+
+from daft.execution.execution_step import PartitionTask
+from daft.runners import ray_metrics
+
+if TYPE_CHECKING:
+    from daft import ResourceRequest
+    from daft.daft import PyDaftExecutionConfig
+    from daft.execution.physical_plan import MaterializedPhysicalPlan
+
+logger = logging.getLogger(__name__)
+
+# We add the trace by default to the latest session logs of the Ray Runner
+# This lets us access our logs via the Ray dashboard when running Ray jobs
+DEFAULT_RAY_LOGS_LOCATION = pathlib.Path("/tmp") / "ray" / "session_latest" / "logs"
+DEFAULT_DAFT_TRACE_LOCATION = DEFAULT_RAY_LOGS_LOCATION / "daft"
+
+# IDs and names for the visualized data
```

Comment:
These should not be in the global namespace and should be class constants inside of `RunnerTracer` instead since they are only used there.